src/client/pages/Section/DataTable/Table/RowData/Cell/hooks/useOnChange/useOnChangeNodeValue.ts

```diff
@@ -0,0 +1,167 @@
+import { useCallback } from 'react'
+import { useTranslation } from 'react-i18next'
+
+import { Objects } from 'utils/objects'
+
+import { CountryIso } from 'meta/area'
+import { Assessment } from 'meta/assessment/assessment'
+import { Cycle } from 'meta/assessment/cycle'
+import { VariableCache } from 'meta/assessment/metaCache'
+import { AssessmentMetaCaches } from 'meta/assessment/metaCaches'
+import { NodeValue } from 'meta/assessment/node'
+import { RowCache } from 'meta/assessment/rowCache'
+import { RecordAssessmentDatas } from 'meta/data'
+import { ExpressionEvaluator } from 'meta/expressionEvaluator'
+
+import { useAppDispatch } from 'client/store'
+import { useAssessment, useCycle } from 'client/store/assessment'
+import { DataActions } from 'client/store/data'
+import { useSection } from 'client/store/metadata'
+import { useCountryIso } from 'client/hooks'
+
+import { OnChangeNodeValue, Props } from './types'
+import { usePersistSanitizedValue } from './usePersistSanitizedValue'
+
+const getAffectedNodes = (
+  props: Props & {
+    assessment: Assessment
+    cycle: Cycle
+    countryIso: CountryIso
+    value: NodeValue
+  }
+): Array<VariableCache> => {
+  const { assessment, col, countryIso, cycle, data, row, sectionName, table, value } = props
+
+  const tableName = table.props.name
+  const { variableName } = row.props
+  const { colName } = col.props
+
+  // Future data contains the updated value and is used to evaluate the enablers
+  const futureData = RecordAssessmentDatas.updateDatum({
+    assessmentName: assessment.props.name,
+    cycleName: cycle.name,
+    countryIso,
+    tableName,
+    colName,
+    variableName,
+    data: Objects.cloneDeep(data),
+    value,
+  })
+
+  const enablerDependants = AssessmentMetaCaches.getEnablersDependants({
+    assessment,
+    cycle,
+    tableName,
+    variableName,
+  })
+
+  return enablerDependants.filter((dep) => {
+    const depRow = table.rows?.find((r) => r.props.variableName === dep.variableName)
+    const depCol = depRow?.cols?.find((c) => c.props.colName === dep.colName)
+    if (!depRow || !depCol) return false
+    const enableIf = depCol.props.enableIf?.[cycle.uuid]
+    if (!enableIf) return false
+
+    const rowCache: RowCache = {
+      ...depRow,
+      tableName,
+      sectionName,
+    }
+
+    const enabled = ExpressionEvaluator.evalFormula<boolean>({
+      assessment,
+      assessments: { [assessment.props.name]: assessment },
+      countryIso,
+      cycle,
+      data: futureData,
+      colName: depCol.props.colName,
+      row: rowCache,
+      formula: enableIf,
+    })
+
+    const depValue = RecordAssessmentDatas.getNodeValue({
+      assessmentName: assessment.props.name,
+      cycleName: cycle.name,
+      data,
+      countryIso,
+      tableName,
+      colName: depCol.props.colName,
+      variableName: depRow.props.variableName,
+    })
+
+    return !enabled && depValue && depValue.raw
+  })
+}
```

Comment:
Main change 2 (called by main change 1)

(Rest of the changes is just splitting the useOnChange to separate files)